### ISSUE TITLE
[a11y]: Contained List [with layer] - is layered considered a nested list?

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#13947]
### Package

@carbon/react

### Browser
FF & Safari & VO

### Package version

carbon/react@1.30.0

### Automated testing tool and ruleset
Manual QA

### Assistive technology
JAWS & VO

### Description

Is this a nested list?

### WCAG 2.1 Violation
n/a

### Reproduction/example
Record `Number 37` in the below table:
https://airtable.com/shrFsaeHhImOkuVNR

### Steps to reproduce

If this is supposed to be A nested list, The semantics should be programmatically conveyed

Team needs to confirmed whether this is A nested list?

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
imo the use case here is slim. conveying these semantics (which yes, is similar to a nested list) would be beneficial for non-sighted developers building with Carbon. but we could explore if there is an easy way to convey these semantics in the stories.

**Comment by andreancardona** [comment]
Here is a similar issue: https://github.com/carbon-design-system/carbon/issues/13980

**Comment by tw15egan** [comment]
I think we need to have these `with Layer` stories removed from the a11y testing. These are supposed to just test + verify that the layer tokens are working properly without any context. I don't think `Layer` needs to have any semantic meaning, it is a decorative change that (I feel) would only add complexity to a screenreader output. These stories are not indicative of real-world usage, and I would like to see more use cases before doing anything with this. Let me know if I'm missing something, but I think this and #13980 should be closed.
